### PR TITLE
fix: fix headers and abort signals

### PR DESCRIPTION
**Pull request by achingbrain**
Adds tests for the same.

Depends on:

- [x] https://github.com/ipfs/aegir/pull/553

### CODE REVIEW

File: src/http.js [hunk 1]
```diff
@@ -88,18 +97,7 @@ class HTTP {
...
-    this.abortController = new AbortController()
-
-    if (this.opts.signal) {
-      this.opts.signal.addEventListener('abort', () => {
-        this.abortController.abort()
-      })
-    }
-
-    this.opts.signal = this.abortController.signal
+    this.opts.headers = options.headers
```
Comment (hugomrdias) [inline src/http.js:100]: this line can probably be removed

File: src/http.js [hunk 2]
```diff
@@ -32,24 +33,32 @@ const timeout = (promise, ms, abortController) => {
...
+    }, ms)
+
     const after = (next) => {
       return (res) => {
         clearTimeout(timeoutID)
-        const time = Date.now() - start
 
-        if (time >= ms) {
-          abortController.abort()
+        if (timedOut()) {
```
Comment (hugomrdias) [inline src/http.js:54]: why are we checking for timeout on resolve and reject, shouldn't it be aborted already ?

`after` shoudn't call resolve nor reject if already aborted

Comment (achingbrain) [reply src/http.js:54]: If the event loop is busy and your timeout is very, very small, it's possible to resolve a promise before the timeout fires, even though for the observer, the amount of time specified for the timeout has passed.  This is mostly evident in weird test behaviour.

File: src/http.js (hunk 1 above)
Comment (achingbrain) [reply src/http.js]: I've removed it.

File: src/http.js (hunk 2 above)
Comment (achingbrain) [reply src/http.js:54]: > `after` shoudn't call resolve nor reject if already aborted

It'll only call reject again if the timeout was reached - we could inspect `abortController.signal.aborted` for completeness but you can't reject a promise twice so it shouldn't hurt.

**Comment by hugomrdias** [comment]
can you update aegir dep to merge this ?

**Comment by achingbrain** [comment]
I think aegir still needs a release with ipfs/aegir#553 in it.

**Pull request merged**